src/modules/fingerprint_acl/fp_acl_file.c

```diff
@@ -55,6 +55,14 @@ fp_acl_db_status fp_acl_file_read_file(FILE* aclFile, struct fp_mem_state* acl)
         READ_FORWARD_U8(acl->users[i].psk.hasValue, ptr);
         READ_FORWARD_MEM(acl->users[i].psk.value.data, ptr, PSK_LENGTH);
 
+#if NABTO_ENABLE_FCM_TOKEN_STORAGE
+        READ_FORWARD_U8(acl->users[i].fcmTok.hasValue, ptr);
+        READ_FORWARD_MEM(acl->users[i].fcmTok.value.data, ptr, FCM_TOKEN_LENGTH);
+#else
+        // Skip
+        ptr += 1 + FCM_TOKEN_LENGTH;
```

Comment:
If FCM token storage is disabled then we still use the memory? why let it be optional then?

Reply:
Tokens are always stored in the persistence.bin file, but having the `NABTO_ENABLE_FCM_TOKEN_STORAGE` set to false means that the `fp_acl_user` struct (which is what's actually stored in memory) will not have the 255 byte buffer, so less memory is used.

The persistence.bin file will always have tokens stored, when a user record is loaded from the file, the token is included in the file buffer. This skip jumps across the token part, thereby not loading it into the fp_acl_user struct (which won't have space for it anyway in the case that the token storage flag is set to false).

Unless I read wrong into it, this means that tokens are loaded from the file and then freed almost immediately afterwards.